lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -522,87 +506,117 @@ lookupPrivateKey keyFrom addr =
-            (Cardano.WithdrawalsShelley $ SL.Wdrl wdrls)
-            Nothing -- Update
-            Nothing -- Metadata hash
-    in
-        unsigned
+            (toCardanoTxIn . fst <$> CS.inputs cs)
+            (map toCardanoTxOut $ CS.outputs cs)
 
 mkWithdrawals
-    :: forall (n :: NetworkDiscriminant). (Typeable n)
-    => Proxy n
+    :: NetworkId
     -> ChimericAccount
     -> Word64
-    -> Map (SL.RewardAcnt TPraosStandardCrypto) SL.Coin
-mkWithdrawals proxy (ChimericAccount keyHash) amount
+    -> [(Cardano.StakeAddress, Cardano.Lovelace)]
+mkWithdrawals networkId acc amount
     | amount == 0 = mempty
-    | otherwise = Map.fromList
-        [ ( SL.RewardAcnt (fromNetworkDiscriminant proxy) keyHashObj
-          , SL.Coin $ fromIntegral amount
-          )
-        ]
+    | otherwise   = [ (stakeAddress, toCardanoLovelace $ Coin amount) ]
   where
-    keyHashObj = SL.KeyHashObj $ SL.KeyHash $ Hash.UnsafeHash keyHash
+    cred = toCardanoStakeCredential acc
+    stakeAddress = Cardano.makeStakeAddress networkId cred
 
 -- NOTE: The (+7200) was selected arbitrarily when we were trying to get
 -- this working on the FF testnet. Perhaps a better motivated and/or
 -- configurable value would be better.
-defaultTTL ::  SlotNo -> SlotNo
-defaultTTL slot = slot + 7200
+defaultTTL :: SlotNo -> SlotNo
+defaultTTL = (+ 7200)
 
 mkShelleyWitness
-    :: SL.TxBody TPraosStandardCrypto
+    :: Cardano.TxBody Cardano.Shelley
     -> (XPrv, Passphrase "encryption")
-    -> SL.WitVKey TPraosStandardCrypto 'SL.Witness
-mkShelleyWitness body (prv, pwd) =
-    SL.WitVKey key sig
+    -> Cardano.Witness Cardano.Shelley
+mkShelleyWitness body key =
+    Cardano.makeShelleyKeyWitness body (unencrypt key)
   where
-    sig = SignedDSIGN
-        $ fromMaybe (error "error converting signatures")
-        $ rawDeserialiseSigDSIGN
-        $ serialize' (SL.hashTxBody body) `signWith` (prv, pwd)
-
-    key = SL.VKey
-        $ VerKeyEd25519DSIGN
-        $ unsafeMkEd25519
-        $ toXPub prv
-
-signWith
-    :: ByteString
-    -> (XPrv, Passphrase "encryption")
-    -> ByteString
-signWith msg (prv, pass) =
-    CC.unXSignature . CC.sign pass prv $ msg
-
-unsafeMkEd25519 :: XPub -> Ed25519.PublicKey
-unsafeMkEd25519 =
-    throwCryptoError . Ed25519.publicKey . xpubPublicKey
+    unencrypt (xprv, pwd) = Cardano.WitnessPaymentExtendedKey
+        $ Cardano.PaymentExtendedSigningKey
+        $ Crypto.HD.xPrvChangePass pwd BS.empty xprv
 
 mkByronWitness
-    :: SL.TxBody TPraosStandardCrypto
-    -> ProtocolMagic
+    :: Cardano.TxBody Cardano.Shelley
+    -> Cardano.NetworkId
     -> Address
     -> (XPrv, Passphrase "encryption")
-    -> SL.BootstrapWitness TPraosStandardCrypto
-mkByronWitness body protocolMagic addr (prv, Passphrase pwd) =
-    SL.makeBootstrapWitness txHash signingKey addrAttr
+    -> Cardano.Witness Cardano.Shelley
+mkByronWitness (Cardano.ShelleyTxBody txbody _) nw addr encryptedKey =
+    Cardano.ShelleyBootstrapWitness $
```

Comment:
Any reason why we refactored all these actually and no longer use `makeBootstrapWitness` :thinking: ? This code was quite tricky to get right in the first place, so I am still very doubtful about it working correctly now.

Reply:
I ended up changing it back in 59535f54c3d49d98c20d432f19dc60c8f1eef172

Reply:
Yes it was confusion related to `Cardano.Api.Typed.makeShelleyBootstrapWitness`.